coalib/testing/LocalBearTestHelper.py

```diff
@@ -0,0 +1,205 @@
+        :param create_tempfile:  Whether to save lines in tempfile if needed.
+        :param tempfile_kwargs:  Kwargs passed to tempfile.mkstemp().
+        """
+        assert isinstance(self, unittest.TestCase)
+        self.assertIsInstance(local_bear,
+                              LocalBear,
+                              msg="The given bear is not a local bear.")
+        self.assertIsInstance(lines,
+                              (list, tuple),
+                              msg="The given lines are not a list.")
+        self.assertIsInstance(results,
+                              list,
+                              msg="The given results are not a list.")
+
+        with prepare_file(lines, filename,
+                          force_linebreaks=force_linebreaks,
+                          create_tempfile=create_tempfile,
+                          tempfile_kwargs=tempfile_kwargs) as (file, fname), \
+                execute_bear(local_bear, fname, file) as bear_output:
+            msg = ("The local bear '{}' doesn't yield the right results. Or the"
+                   " order may be wrong.".format(local_bear.__class__.__name__))
+            if not check_order:
+                self.assertEqual(sorted(bear_output), sorted(results), msg=msg)
+            else:
+                self.assertEqual(bear_output, results, msg=msg)
+
+
+def verify_local_bear(bear,
+                      valid_files,
+                      invalid_files,
+                      filename=None,
+                      settings={},
+                      force_linebreaks=True,
+                      create_tempfile=True,
+                      timeout=None,
+                      tempfile_kwargs={}):
+    """
+    Generates a test for a local bear by checking the given valid and invalid
+    file contents. Simply use it on your module level like:
+
+    YourTestName = verify_local_bear(YourBear, (['valid line'],),
+                                     (['invalid line'],))
+
+    :param bear:             The Bear class to test.
+    :param valid_files:      An iterable of files as a string list that won't
+                             yield results.
+    :param invalid_files:    An iterable of files as a string list that must
+                             yield results.
+    :param filename:         The filename to use for valid and invalid files.
+    :param settings:         A dictionary of keys and values (both string) from
+                             which settings will be created that will be made
+                             available for the tested bear.
+    :param force_linebreaks: Whether to append newlines at each line
+                             if needed. (Bears expect a \\n for every line)
+    :param create_tempfile:  Whether to save lines in tempfile if needed.
+    :param timeout:          The total time to run the test for.
+    :param tempfile_kwargs:  Kwargs passed to tempfile.mkstemp() if tempfile
```

Comment:
Backticks (here and other places in the docs (if any), don't want to clutter the comments with the same thing)